add UnusedImports and RedundantImports

mattnworb opened the pull request:
We don't have warnings for either of these two today, although it seems like it would be useful. 

Was there a conscious reason for leaving these out?

mattnworb committed:
add UnusedImports and RedundantImports

pettermahlen commented on src/main/resources/spotify_checks.xml:
```diff
@@ -50,6 +50,8 @@
             <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
         </module>
         <module name="AvoidStarImport"/>
+	<module name="RedundantImport"/>
```
Indentation?

mattnworb replied:
this is what i get for trying to edit xml in the github UI

pettermahlen reviewed (commented):
Looks good to me, except the indentation is different according to the diff.

mattnworb committed:
fix: consistent indenting of import rules

last commit accidentally used tabs instead of spaces, thanks for nothing github editing UI

davidxia reviewed (approved):
I guess they weren't in google's checkstyle (which has recently been udpated).
